3.0.2 (#9)
ocient/pyocient_release: README.md

# Pyocient Release Notes

## 3.0.2

> - The `TypeCodes` class has been changed to a python `IntEnum`

Reviewer: ```suggestion
- The `TypeCodes` class has been changed to a Python® `IntEnum`.
```

> - GIS types (STPoint, STLinestring, STPolygon) have been renamed from _STPoint, _STLinestring, _STPolygin

Reviewer: ```suggestion
- GIS types  _STPoint, _STLinestring, and _STPolygon have been renamed to STPoint, STLinestring, and STPolygon, respectively.
```

> - GIS types (STPoint, STLinestring, STPolygon) have been renamed from _STPoint, _STLinestring, _STPolygin

Reviewer 2: @cj81499 is this a breaking change to move a "private" type to a public name? I don't think so?

> - The `user` and `password` fields have been removed from the `Connection` object and are no longer stored in the object

Reviewer: ```suggestion
- The `user` and `password` fields have been removed from the `Connection` object. This object no longer stores these fields.
```

> - A `security_token` from another connection can be passed in to the `connect` function

Reviewer: ```suggestion
- You can pass the `security_token` from another connection to the `connect` function.
```

> - The `Session` object has been removed

Reviewer: ```suggestion
- The `Session` object has been removed.
```

> - The `custom_type_to_json` function has been moved into the pyocient API to allow consistent conversion of resultset data to JSON format

Reviewer: ```suggestion
- The `custom_type_to_json` function has been moved into the pyocient API to allow the consistent conversion of the result set to the JSON format.
```

> - The `patch` portion of the pyocient version is now sent to the database and will be logged there

Reviewer: ```suggestion
- The `patch` portion of the pyocient version is sent to and logged in the database.
```

> - The `--nooutput` command line parameter has been fixed to not throw a python exception

Reviewer: ```suggestion
- The `--nooutput` command-line parameter does not throw a Python exception.
```

> - An error reporting `local variable 'return_code' referenced before assignment` has been fixed

Reviewer: ```suggestion
- The `local variable 'return_code' referenced before assignment` error has been fixed.
```

> - A `--rows` argument to the command line program has been added to allow loading resultset data in batches

Reviewer: ```suggestion
- Added the `--rows` argument to the command-line program to load the result set in batches.
```

> - An alpha version of OpenAI "text to SQL" has been added to the command line

Reviewer 2: It's also not strictly OpenAI, it supports Anthropic's API, as well.

> - An alpha version of OpenAI "text to SQL" has been added to the command line

Reviewer 2: Should a new feature like this warrant a minor version bump - 3.1.0?

Reviewer 3: yes

Reviewer 3: What happened to `3.0.0`...? Why are we skipping versions?

Author: because sometimes versions get built and posted here: https://artifactory.corp.ocient.com/artifactory/tools_pypi/user_dboutcher_pyocient_version/pyocient/ that don't get published publicly

Reviewer 3: It's **very** strange to me to bump the **public** version for internal-only releases (aka pre-releases). see: https://semver.org/spec/v2.0.0.html#spec-item-9

Author: perhaps when the tools team rewrites the python packaging to be more Stoyan friendly, they can additionally make it support non-numeric package versions

# Pyocient Release Notes

## 3.0.3

- The `TypeCodes` class has been changed to a Python® `IntEnum`.

> - GIS types  _STPoint, _STLinestring, and _STPolygon have been renamed to STPoint, STLinestring, and STPolygon, respectively.

Reviewer: ```suggestion
- GIS types _STPoint, _STLinestring, and _STPolygon have been renamed to STPoint, STLinestring, and STPolygon, respectively.
```
